src/omnicore/rpctxobject.cpp

```diff
@@ -256,6 +256,7 @@ void populateRPCTypeTradeOffer(CMPTransaction& omniObj, Object& txobj)
 
     // Check levelDB to see if the amount for sale has been amended due to a partial purchase
     // TODO: DEx phase 1 really needs an overhaul to work like MetaDEx with original amounts for sale and amounts remaining etc
+    // TODO: if the amount of the transaction is not used here, then the BTC amount should be recalculated (?)
```

Comment:
YES! :)

It's not backwards compatible though, and needs `legacy::` handling. There are historical transactions which would otherwise be invalid. (Same goes for two active DEx offers [MSC/TMSC] at the same time, which I checked, when I created #167))